spawnpool.py

```diff
@@ -205,25 +204,29 @@ def summarize(message, list):
             self._heart_beating = False
 
     @gen.coroutine
-    def _launch_container(self, path=None, enpool=True):
+    def _launch_container(self, user=None, enpool=True):
         '''Launch a new notebook server in a fresh container, register it with the proxy, and
         add it to the pool.'''
 
-        if path is None:
-            path = user_prefix()
+        if user is None:
+            user = new_user()
 
-        # This must match self.name_pattern or Bad Things will happen.
+        path = "user/" + user
```

Comment:
A thought: users may be able to provide names that include characters that aren't valid in container names (which must match [`/?[a-zA-Z0-9_-]+`](https://docs.docker.com/reference/api/docker_remote_api_v1.16/#create-a-container). Do we defend against this?

If we don't it's probably worth an issue/PR on its own rather than fixing it here, though.